src/v/s3/client.h

```diff
@@ -133,6 +176,8 @@ class request_creator {
 /// S3 REST-API client
 class client {
 public:
+    struct no_response {};
```

Comment:
Note for reviewer: this empty struct is used in the return type of client member function that used to return a `void` future. It allows for the callers of `client` to use consistent error handling via `result<T, error_outcome>` (which is not the case for the `result<void, error_outcome>` alternative.